cmake/HalideGeneratorHelpers.cmake

```
@@ -414,6 +414,9 @@ function(add_halide_library TARGET)
         set_target_properties("${TARGET}" PROPERTIES
                               POSITION_INDEPENDENT_CODE ON
                               LINKER_LANGUAGE CXX)
+        # Silence many useless warnings in generated C++ code compilation
+        target_compile_options("${TARGET}" PRIVATE
+            $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wno-psabi>)
```

This needs to be guarded by `Halide_NO_DEFAULT_FLAGS` in case a newer compiler removes the psabi warning and the flag becomes invalid.
```suggestion
        if (NOT Halide_NO_DEFAULT_FLAGS)
            # Silence many useless warnings in generated C++ code compilation
            target_compile_options("${TARGET}" PRIVATE
                $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wno-psabi>)
        endif ()
```